### ISSUE TITLE
Failed to work after 5000 lines of code

### ISSUE THREAD

**Issue by ChickenFood** [CoenraadS/BracketPair#154]
Language: C
Problem: As stated in the title. Any bracket pair after line 5000 fails to be colored

**Comment by lilnayners** [comment]
I am seeing the same thing happen in .less files, as well

**Comment by CoenraadS** [comment]
I hard coded this to prevent performance degradation, but I can move it to
a setting

**Comment by bencun** [comment]
> I hard coded this to prevent performance degradation, but I can move it to
a setting

Well, in all honesty, there are still a lot of performance issues. I was forced to uninstall the extension (although I love it and keep using it at home) seeing that my work laptop with i5-5200U cannot handle it properly in any file larger than 1k lines. It gets exponentially worse as the files become larger. Our unit test files regularly exceed 2-3k lines of code and the extension simply kills the performance on that machine to a point that when editing the code that's in a more broad scope I have to wait up to a second or two for the inputs to register.

Maybe it would be nice to have a user preference where the user could limit the scope of the extension? For example, I usually only need the bracket highlighting for the function that I'm currently editing and a scope or two outside of that function.
Also, is it possible to improve the performance by detecting the collapsed code blocks and skip the processing of those?

These are just some random suggestions and thoughts. I didn't want to open a new issue seeing that I know that improving performance on an extension like this simply has it's limits but I still wanted to give you my two cents on the current status of the performance.

**Comment by commonterry** [comment]
Yes - please move this to a setting as I am experiencing it as well. Thanks!
